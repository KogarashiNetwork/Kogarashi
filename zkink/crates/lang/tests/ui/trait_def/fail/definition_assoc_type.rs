use ink_lang as ink;

#[ink::trait_definition]
pub trait TraitDefinition {
    type Type;

    #[ink(message)]
    fn message(&self);
}

fn main() {}
