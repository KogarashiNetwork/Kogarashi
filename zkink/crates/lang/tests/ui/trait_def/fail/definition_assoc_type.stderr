error: associated types in ink! trait definitions are not supported, yet
 --> tests/ui/trait_def/fail/definition_assoc_type.rs:5:5
  |
5 |     type Type;
  |     ^^^^^^^^^^
